lib/optimize/ConcatenatedModule.js

```diff
@@ -168,17 +168,16 @@ const getFinalName = (
 	}
 };
 
-const getSymbolsFromScope = (s, untilScope) => {
-	const allUsedNames = new Set();
+const addScopeSymbols = (nameSet, scopeSet, s, untilScope) => {
 	let scope = s;
 	while (scope) {
+		if (scopeSet.has(scope)) break;
```

Comment:
Check scope in `while`?
```js
while (scope) {
		if (scopeSet.has(scope)) break;
```
=>
```js
while (scope && !scopeSet.has(scope)) {
```

Reply:
I prefer how Tobias wrote it. It is more expressive and easier to understand how it works.